File: OOPS_LAB/Program_2/date.py
class DATE:
    def __init__(self,dd,mm,yyyy):
        self.day=dd
        self.month=mm
        self.year=yyyy
    
    def validateDate(self):
        if self.day <= 0 or self.day>=32:
            return False
        
        if self.month<=0 or self.month>=13:
            return False
        
        if self.year<=0 or self.year>=9999:
            return False
        
        if self.month==2:
            if self.year%4==0:
                if self.day>=30:
                    return False
            else:
                if self.day>=29:
                    return False
                
        return True
    
    def display(self):
        print(f"{self.day}/{self.month}/{self.year}")


if __name__=="__main__":
    day1=DATE(29,2,2024)
    if day1.validateDate():
        print("Date is valid")
        day1.display()
    else:
        print("Date is invalid")

File: OOPS_LAB/Program_2/date.js
class DATE{
    constructor(dd=1,mm=1,yyyy=2026)
    {
        this.dd=dd;
        this.mm=mm;
        this.yyyy=yyyy
    }

    validateDate()
    {
        if(this.dd<=0 || this.dd>=32) return false;
        if(this.mm<=0 || this.mm>=13) return false;
        if(this.yyyy<=0 || this.yyyy>=9999) return true;
        if(this.mm==2)
        {
            if(this.yyyy%4==0)
            {
                if(this.dd>=30) return false;
            }
            else{if(this.dd>=29) return false;}
        }
        return true;
    }

    display()
    {
        console.log(`${this.dd}/${this.mm}/${this.yyyy}`);
    }
}

const day1= new DATE(41,3,2026);
if(day1.validateDate())
{
    console.log(`Date is Valid`);
    day1.display();
}
else{
    console.log(`Date is invalid`);
}